File: SDKs/googleapis.com/vision/v1p1beta1/go/pkg/models/shared/googlecloudvisionv1p1beta1inputconfig.go
package shared

// GoogleCloudVisionV1p1beta1InputConfig
// The desired input location and metadata.
type GoogleCloudVisionV1p1beta1InputConfig struct {
	Content   *string                              `json:"content,omitempty"`
	GcsSource *GoogleCloudVisionV1p1beta1GcsSource `json:"gcsSource,omitempty"`
	MimeType  *string                              `json:"mimeType,omitempty"`
}

File: SDKs/googleapis.com/vision/v1p1beta1/typescript/src/sdk/models/shared/googlecloudvisionv1p1beta1inputconfig.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { GoogleCloudVisionV1p1beta1GcsSource } from "./googlecloudvisionv1p1beta1gcssource";



// GoogleCloudVisionV1p1beta1InputConfig
/** 
 * The desired input location and metadata.
**/
export class GoogleCloudVisionV1p1beta1InputConfig extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=content" })
  content?: string;

  @SpeakeasyMetadata({ data: "json, name=gcsSource" })
  gcsSource?: GoogleCloudVisionV1p1beta1GcsSource;

  @SpeakeasyMetadata({ data: "json, name=mimeType" })
  mimeType?: string;
}
